Fix remarks, insert widoco

lecoqlibre opened the pull request.

lecoqlibre committed:
Rewrite Abstract, introduction and title

lecoqlibre committed:
Update specs/solid-indexing/index.html

Typo

Co-authored-by: Pierre-Antoine Champin <github-100614@champin.net>

lecoqlibre committed:
Update specs/solid-indexing/index.html

Rephrase

Co-authored-by: Pierre-Antoine Champin <github-100614@champin.net>

lecoqlibre committed:
Update specs/solid-indexing/index.html

Rephrase

Co-authored-by: Pierre-Antoine Champin <github-100614@champin.net>

lecoqlibre committed:
Use <#thing> instead of :thing in the examples

lecoqlibre committed:
Fix terminology index keyword

lecoqlibre committed:
Replace targeting by referring to

lecoqlibre committed:
Insert Widoco section and change 3.1 index description

lecoqlibre committed:
Merge branch 'specs-dev' of github.com:Wimmics/solid-start into specs-dev

lecoqlibre committed:
Merge branch 'specs' into specs-dev

pchampin commented on specs/solid-indexing/index.html:
```diff
@@ -88,7 +130,219 @@ <h3>Namespaces</h3>
   </section>
 
   <section>
```
some intro text at the top of the vocabulary section would be nice... Currently it is very abrupt.

lecoqlibre replied:
Fixed in https://github.com/Wimmics/solid-start/pull/47/commits/b71e281de8267a11f748a96e324bf6176435a126

pchampin commented on specs/solid-indexing/index.html:
```diff
@@ -102,13 +356,16 @@ <h2>Indexes</h2>
+      <p>An index is a RDF document [[RDF11-CONCEPTS]] of type <code>idx:Index</code>. It contains entries of type <code>idf:IndexEntry</code>, each one 
+        being linked to a shape with the <code>idx:hasShape</code> predicate.</p>
+        
+      <p>An entry MUST refer to either an instance or to a sub index with the respective predicates <code>idx:hasTarget</code> and <code>idx:hasSubIndex</code>.</p>
```
```suggestion
      <p>An entry MUST refer to either resources matching the shape  or to another index referring to such resources, with predicate <code>idx:hasTarget</code> or <code>idx:hasSubIndex</code> respectively.</p>
```

pchampin replied:
or, if you don't want to allow multiple values for hasTarget/hasSubIndex.

```suggestion
      <p>An entry MUST refer to a resource matching the shape  or to another index referring to such resources, with predicate <code>idx:hasTarget</code> or <code>idx:hasSubIndex</code> respectively.</p>
```

pchampin commented on specs/solid-indexing/index.html:
```diff
@@ -44,7 +86,7 @@ <h3>Terminology</h3>
 
       <p>
         <dfn data-lt="indexes">Index</dfn>
-        A list of keywords with pointers to where further information about the keyword is found. An index summarizes or
+        A list of keyword with pointers to where further information about the keyword is found. An index summarizes or
         characterizes the content and its location.
```
Why remove the plural??
Anyway, I didn't notice this definition before, but I don't find it satisfying. The notion of keyword does not match what we are doing (or only partially).

I can not make a suggestion here (this impacts non-modified lines), I will make a PR to the PR.

lecoqlibre replied:
This definition was the one used in our different presentations (slides).

lecoqlibre replied:
Fixed in https://github.com/Wimmics/solid-start/pull/47/commits/7c73164dd34d8aaddd7e8e64ca93e83bae74f4e2

pchampin commented on specs/solid-indexing/index.html:
```diff
@@ -102,13 +356,16 @@ <h2>Indexes</h2>
+        being linked to a shape with the <code>idx:hasShape</code> predicate.</p>
+        
+      <p>An entry MUST refer to either an instance or to a sub index with the respective predicates <code>idx:hasTarget</code> and <code>idx:hasSubIndex</code>.</p>
+
+      <p>An instance is...</p>
+
+      <p>A sub index is...</p>
```
```suggestion
```
No need for this with my proposed rewording above.

pchampin commented on specs/solid-indexing/index.html:
```diff
@@ -102,13 +356,16 @@ <h2>Indexes</h2>
 
-      <p>An entry is of type <code>idx:IndexEntry</code> and targets a resource using the <code>idx:hasTarget</code>
-        predicate. The resource the entry is targeting MUST conform to the shape of the index.</p>
+      <p>A shape is...</p>
```
suggest to copy the definition from SHACL (with a reference).

Maybe this is the place to explain that this spec recommends a very small subset of shapes.

Interestingly, we could write a SHACL shape of the SHALC shapes we allow :-)

lecoqlibre committed:
Add intro text at the top of the vocabulary section

pchampin committed:
fix invalid turtle in example 4

pchampin committed:
add missing prefixes (they are used in examples)

+ the RDF namespace is the one for the core vocabulary,
not for RDFS

pchampin committed:
better (IMO) definition of index

(loosely based on https://en.wikipedia.org/wiki/Database_index)

pchampin committed:
improve ontology

- better labels
- prefix are hermit-friendly
- hasTarget is no longer functional
- add axioms to make hasTarget and hasSubIndex exclusive
- add axiom for hasCount to require hasSubIndex

lecoqlibre committed:
Allow only one value for hasTarget and hasSubIndex

We need only one value to use hasCount with hasSubIndex.

Co-authored-by: Pierre-Antoine Champin <github-100614@champin.net>
